atc/steps.go

```diff
@@ -158,13 +167,13 @@ type StepConfig interface {
 }
 
 // StepWrapper is an optional interface for step types that is implemented by
-// steps that wrap other steps (e.g. hooks like `on_success`, `timeout`, etc.)
+// steps that wrap/modify other steps (e.g. hooks like `on_success`, `timeout`, etc.)
 type StepWrapper interface {
 	// Wrap is called during (Step).UnmarshalJSON whenever an 'inner' step is
 	// parsed.
 	//
-	// Modifier step types should implement this by assigning to an internal
-	// field that has a `json:"-"` struct tag.
+	// Modifier step types should implement this by assigning a `json:"-"` tag to
+	// an internal field of type StepConfig in their struct.
```

Comment:
This re-wording says something different than what was intended. The original statement was that modifiers should implement the `Wrap` method by assigning the given `StepConfig` to a field of the struct, and that same field should have a `json:"-"` tag so that it is ignored during regular unmarshalling.

This comment suggests that the `json:"-"` tag should be added to the field, but doesn't say that the `Wrap` method should assign it (which is arguably the more important part).

Reply:
I reverted the comment to the original one and then made some changes to it. let me know what you think!